src/Voron/Impl/LowLevelTransaction.cs

```diff
@@ -419,6 +419,19 @@ public Page GetPage(long pageNumber)
             return GetPageInternal(pageNumber);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Page GetPage(long pageNumber, PagerRef pagerRef)
+        {
+            if (_disposed)
+                ThrowObjectDisposed();
+
+            // PERF: ReadOnly pages will not have a PagerRef, so we can still use the cache. 
+            if (_pageLocator.TryGetReadOnlyPage(pageNumber, out Page result))
```

Comment:
I am still trying to understand the semantics of EnsureMapped. When is required and when it is not. Ref though won't be populated unless we go deep into the storage.